### ISSUE TITLE
MODx _log() method 'target' issue

### ISSUE THREAD

**Issue by bertoost** [modxbot/migrate#4013]
bertoost created Redmine issue ID 4013

For a shop project I have made and script that imports products from the system in the 'reallife' store. For this script I have set the log target to an own directory and entered an custom filename. At the top of the script I have put this code:

$modx = new modX();  
$modx->initialize('web');
$modx->setLogLevel(modX::LOG_LEVEL_INFO);
$modx->setLogTarget(array(
   'target' => 'FILE',
   'options' => array(
       'filepath' => dirname(**FILE**).'/logs/',
       'filename' => 'import.' . strftime('%Y%m%dT%H%M%S')
    )
));

After some search I found the overruling _log() method in the modx.class.php. At the top of this method the 'target' is figured out and $target is containing 'FILE'. But at the bottom of the method the parent::_log() is called and $target isn't an array anymore, but a string.

In the parent::_log() method the target isn't an array anymore what causes that my settings for the log target don't work. The modX::_log() $target parameter must be given trough the parent::_log() method without changes!

To test if I am right; I have add this line at the top of modX::_log() method

$orgTarget = $target;

And where the parent::_log() is called I send the $orgTarget instead of $target and now my custom log file is created!

**Comment by bertoost** [comment]
bertoost submitted:

For now I have extend MODx and overruled the _log() method and call the xPDO::_log() instead of parent::_log(). This is working for me right now.

<?php

class MyMODx extends modX {

```
public function _log($level, $msg, $target= '', $def= '', $file= '', $line= '') {

    xPDO::_log($level, $msg, $target= '', $def= '', $file= '', $line= '');
}
```

}

?>

**Comment by opengeek** [comment]
opengeek submitted:

Fixed for 2.1.0-rc2
